serverless-plugin/alarms-alb-target-group.js

```diff
@@ -0,0 +1,206 @@
+'use strict'
+
+const { makeResourceName, getStatisticName } = require('./util')
+/**
+   * Given CloudFormation syntax for an Target Group, derive CloudFormation syntax for
+   * the LoadBalancer LogicalId name
+   *
+   * @param  loadBalancerResources  syntax for an Load Balancer Application
+   * @returns CloudFormation syntax for the Load Balancer Resources
+   */
+function resolveLoadBalancerLogicalIdName (loadBalancerResources) {
+  for (const key in loadBalancerResources) {
+    if (loadBalancerResources[key].Type === 'AWS::ElasticLoadBalancingV2::LoadBalancer') {
+      return key
+    }
+  }
+}
+
+/**
+ * @param {object} albTargetAlarmConfig The fully resolved alarm configuration
+ */
+module.exports = function ALBlarms (albTargetAlarmConfig, context) {
+  return {
+    createALBTargetAlarms
+  }
+
+  /**
+   * Add all required Application Load Balancer alarms for Target Group to the provided CloudFormation template
+   * based on the resources found within
+   *
+   * @param {CloudFormationTemplate} cfTemplate A CloudFormation template object
+   */
+  function createALBTargetAlarms (cfTemplate) {
+    const loadBalancerResources = cfTemplate.getResourcesByType(
+      'AWS::ElasticLoadBalancingV2::LoadBalancer'
+    )
+
+    const targetGroupResources = cfTemplate.getResourcesByType(
+      'AWS::ElasticLoadBalancingV2::TargetGroup'
+    )
+    for (const [targetGroupResourceName, targetGroupResource] of Object.entries(targetGroupResources)) {
+      const loadBalancerName = resolveLoadBalancerLogicalIdName(loadBalancerResources)
```

Comment:
this function (`resolveLoadBalancerLogicalIdName`) is finding a logical ID but the assigned variable is called `loadBalancerName`, so could lead to confusion.

Reply:
Aside from that issue, it looks like this function just finds the first Load Balancer in the CF template. What happens if the Load Balancer is not associated with this target group?

Reply:
Yes, that is right. In that case even though Load Balancer is not associated with the right target group CF template will read the first ALB anyway and it is not what we want..